From 7a447caf05ec70c35c9df4d5211b9dd3828e6b6a Mon Sep 17 00:00:00 2001
From: Natanael Copa <ncopa@alpinelinux.org>
Date: Mon, 21 Oct 2013 13:48:23 +0000
Subject: [PATCH] buildrepo: exit early if listing of packages for purging
 fails

Otherwise we might end up purging more packages than expected if an
APKBUILD has syntax errors
---
 buildrepo.in | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/buildrepo.in b/buildrepo.in
index a37ab8d..dcba9b6 100755
--- a/buildrepo.in
+++ b/buildrepo.in
@@ -146,7 +146,7 @@ do_build() {
 		local purgefiles
 		cd "$repodir/$repo/$CARCH" || return 1
 		trap 'rm -f "$tmp"; exit 1' INT
-		( listpackages "$1") >$tmp
+		( listpackages "$1") >$tmp || return 1
 		purge=$(ls *.apk 2>/dev/null | grep -v -w -f $tmp)
 		if [ -n "$purge" ]; then
 			rm -f $purge
-- 
1.8.4.1

